R/distances.R

```diff
@@ -419,41 +430,104 @@ distancesMatrix <- function(t.layer, starters = NULL, targets = starters,
   }
 
   if (!inherits(t.layer, "TransitionLayer"))
-    stop("Could not recognise 't.layer' as a TransitionLayer object. Make sure to compile it using the function transitionLayer.\n", call. = FALSE)
+    stop("Could not recognise 't.layer' as a TransitionLayer object. Make",
+         " sure to compile it using the function transitionLayer.\n", 
+         call. = FALSE)
 
   if (!is.null(id.col) && length(id.col) > 1)
     stop("Please provide only one column name in 'id.col'", call. = FALSE)
   if (!is.null(id.col) && is.numeric(id.col))
-    stop("Please refer to the column name in 'id.col', rather than the column index.\n", call. = FALSE)
+    stop("Please refer to the column name in 'id.col', rather than the column",
+         " index.\n", call. = FALSE)
 
   if (actel) {
     message("M: Creating actel-compatible distances matrix."); flush.console()
-    if (!is.null(starters) | !is.null(targets))
-      warning("starters' or 'targets' were set but will be ignored because 'actel' is set to TRUE. Set 'actel' to FALSE to use the 'starters' and 'targets' arguments.", call. = FALSE, immediate. = TRUE)
-    starters <- targets <- loadSpatial()
-    if (!is.null(id.col))
-      warning("id.col' was set but will be ignored because 'actel' is set to TRUE. Set 'actel' to FALSE to use the 'id.col' argument.", call. = FALSE, immediate. = TRUE)
+    
+    if(!is.null(starters) | !is.null(targets)) {
+      # distancesMatrix() crashes if actel = TRUE and starters/targets were 
+      # supplied but do not contain a column called "Standard.name"
+      if(!is.null(starters) & !any(grepl(c("Standard.name"), 
+                                         colnames(starters)))) {
+        event("'starters' is missing column called 'Standard.name'",
+              type = "stop")
+      }
+      
+      if(!is.null(targets) & !any(grepl(c("Standard.name"), 
+                                         colnames(targets)))) {
+        event("'targets' is missing column called 'Standard.name'",
+              type = "stop")
+      }
+    }
+    
+    if (!is.null(starters) & !is.null(targets)) {
+      warning("'starters' and 'targets' were specified while 'actel'",
+              " is set to TRUE. Using 'starters' and 'targets' for distances.", 
+              call. = FALSE, immediate. = TRUE) 
+    }
+    
+    if (!is.null(starters) & is.null(targets)) {
+        warning("'starters' were specified but 'targets' were not while 'actel'",
+              " is set to TRUE. Using 'starters' for 'targets'.", 
+              call. = FALSE, immediate. = TRUE) 
+      
+        targets <- starters
+        }
+    
+    if (is.null(starters) & is.null(targets)) {
+        warning("'starters' and 'targets' were not set while 'actel' is set to",
+              " TRUE. Creating 'starters' and 'targets' from 'spatial.csv' ",
+              " in working directory." , call. = FALSE, immediate. = TRUE)
+        starters <- targets <- loadSpatial() 
+        }
```

Comment:
do we need these three warnings? I think the first two could go away; it's just the function doing what it was ordered to do. The last one could be downgraded to a message. Something perhaps simpler e.g.

```
        message("'starters' and 'targets' were not set.",
                " Creating 'starters' and 'targets' from 'spatial.csv' ",
                " in working directory.")
```

What do you think?

Reply:
@hugomflavio  

My objective was to add more user feedback as to how the function was operating in different situations. While they do not provide critical information to the user, they do provide some insight into how the function is using the data they provided to create the distances matrix. 

I do agree they are not "warnings" and could be messages instead. 

What are your thoughts? 

I can defiantly remove the first two and only retain the last (as a message) if you'd prefer it that way. It really depends how much feedback you want to give to the user about the function's operations.

Reply:
right, makes sense, since it is copying one argument to another. Can we make them messages instead and shorten by removing the actel = TRUE part? By this logic, this part of the code should also have a message, no?

![image](https://github.com/user-attachments/assets/67b613b2-d47c-4145-aa10-a989a3de2332)

Reply:
Great point Hugo. I will add a message there and make those revisions now.